### ISSUE TITLE
Broken update from 0.31.3 to 0.32.4

### ISSUE THREAD

**Issue by dmitriishaburov** [aws/karpenter-provider-aws#5448]
### Description

**Observed Behavior**:

When trying to update Karpenter from 0.31.3 (and any version before) to 0.32.4, pods are failing into CrashLoopBackOff with
```
panic: parsing settings, invalid character 'm' looking for beginning of value

goroutine 1 [running]:
```

**Expected Behavior**:
Update going successfully

**Reproduction Steps** (Please include YAML):
We have following block in our helm chart values:
```
    settings:
      aws:
        tags:
          sometag: somevalue
```
In 0.31.3 and earlier it renders into following configmap:
```
  aws.tags: '{"sometag":"somevalue"}'
```

In 0.32.4 it looks like this:
```
  aws.tags: map[sometag:somevalue]
```

Obviously Karpenter fails to parse that configuration.

**Versions**:
- Chart Version: 0.32.4
- Kubernetes Version (`kubectl version`): v1.26.11-eks-8cb36c9

**Comment by jonathan-innis** [comment]
Yep, looks like there is an issue here with how the helm chart propagated the global tagging starting on this version. I have a fix out for this in #5450.

**Comment by jonathan-innis** [comment]
Closed by #5450. This has now been released with v0.32.5

**Comment by njtran** [comment]
Testing issue: https://github.com/aws/karpenter-provider-aws/issues/4989
